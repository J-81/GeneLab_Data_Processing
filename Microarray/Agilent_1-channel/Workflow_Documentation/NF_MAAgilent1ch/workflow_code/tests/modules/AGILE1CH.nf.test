nextflow_process {

    name "Test Process AGILE1CH"
    script "modules/AGILE1CH.nf"
    process "AGILE1CH"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("/workspace/GeneLab_Data_Processing/Microarray/Agilent_1-channel/Workflow_Documentation/NF_MAAgilent1ch/workflow_code/bin/Agile1CMP.qmd")
                input[1] = file("/workspace/GeneLab_Data_Processing/Microarray/Agilent_1-channel/Workflow_Documentation/NF_MAAgilent1ch/examples/runsheet/GLDS-367_microarray_v0_runsheet.csv")
                input[2] = "agilent_wholegenome_4x44k_v1"
                input[3] = file("https://figshare.com/ndownloader/files/36597120")
                input[4] = "ENSEMBL"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.de, process.out.version).match()
        }

    }

}
