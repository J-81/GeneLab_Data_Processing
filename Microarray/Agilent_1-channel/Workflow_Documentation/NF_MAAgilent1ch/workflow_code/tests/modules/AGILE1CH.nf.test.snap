{
    "Should run without failures": {
        "content": [
            [
                [
                    "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                    "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                    "differential_expression.csv:md5,5df226d4bcc253a207883d4dfa19edbd",
                    "visualization_output_table.csv:md5,7f03c096dad28a3119ed5c43f05abc35",
                    "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9"
                ]
            ],
            [
                "versions.txt:md5,b09b5f76b3dd6b72c41d3f8312de0d64"
            ]
        ],
        "timestamp": "2023-01-17T23:42:40+0000"
    }
}